src/sensors/src/PredictSensorsMeasurements.cpp

```diff
@@ -0,0 +1,185 @@
+#include <iDynTree/Core/SpatialMotionVector.h>
+#include <iDynTree/Sensors/Accelerometer.h>
+#include <iDynTree/Sensors/Gyroscope.h>
+#include <iDynTree/Sensors/Sensors.h>
+#include <iDynTree/Model/Model.h>
+#include <iDynTree/Model/Traversal.h>
+#include <iDynTree/Model/ForwardKinematics.h>
+#include <iDynTree/Model/FreeFloatingState.h>
+#include <iDynTree/Core/SpatialAcc.h>
+#include <map>
+#include <iDynTree/Core/VectorDynSize.h>
+
+
+
+namespace iDynTree {
+PredictSensorsMeasurements::PredictSensorsMeasurements()
+{
+// All local variables, so nothing to do really.
+}
+PredictSensorsMeasurements::~PredictSensorsMeasurements()
+{
+// All local variables, so nothing to do really.
+}
+ 
+bool PredictSensorsMeasurements::operator()(const Model& model,const Traversal& traversal,const iDynTree::FreeFloatingPos& robotPos,const iDynTree::FreeFloatingVel& robotVel,iDynTree::FreeFloatingAcc& robotAcc,const LinAcceleration& gravity,const iDynTree::SensorsList &sensorsList,iDynTree::SensorsMeasurements &predictedMeasurement)
+{
+    
+    bool returnVal = true;
+    // incorporating gravity into the base LinAcceleration
+    iDynTree::LinkPositions linkPos(model);
+    iDynTree::LinkVelArray linkVel(model);
+    iDynTree::LinkAccArray linkAcc(model);
+    
+    
+    iDynTree::AngAcceleration nullAngAccl;
+    iDynTree::SpatialAcc gravityAccl(gravity,nullAngAccl);
+    
+    robotAcc.baseAcc() = robotAcc.baseAcc() - gravityAccl;
+    
+    // calling ForwardPosVelAccKinematics with arguments
+    ForwardPosVelAccKinematics(model,traversal,robotPos,robotVel,robotAcc,linkPos,linkVel,linkAcc);
+    
+    // looping though SensorList    
+    unsigned int numAccl = sensorsList.getNrOfSensors(iDynTree::ACCELEROMETER);
+    unsigned int numGyro = sensorsList.getNrOfSensors(iDynTree::GYROSCOPE);
+    int idx;
+    
+   Accelerometer * accelerometer;
+    Gyroscope * gyroscope;
+    
+    //Iterate through each kind of accelrometer and find its parent. Compute local (classical accelration) 
+    // It is automatically proper acceleration since gravity is incorporated into the base acceleration
+    for(idx = 0; idx<numAccl; idx++)
+    {
+        accelerometer = (Accelerometer *)sensorsList.getSensor(iDynTree::ACCELEROMETER, idx);
+        int parentLinkId = accelerometer->getParentIndex();
+        
+        iDynTree::LinAcceleration measuredAcc, crossProdTerm;
+        iDynTree::Transform link_T_sensor;
+        iDynTree::SpatialAcc linkAccTerm;
+        iDynTree::Twist localVelocity;
+        returnVal = accelerometer->getLinkSensorTransform(link_T_sensor);
+        localVelocity = link_T_sensor*linkVel(parentLinkId);
+        linkAccTerm = link_T_sensor*(linkAcc(parentLinkId));
+        crossProdTerm = (localVelocity.getAngularVec3()).cross(localVelocity.getLinearVec3());
+        measuredAcc = (linkAccTerm).getLinearVec3() + crossProdTerm;       
+        predictedMeasurement.setMeasurement(iDynTree::ACCELEROMETER,idx,measuredAcc);
```

Comment:
I am not a taliban of avoiding code duplication when it make sense, but this is the typical code in which all kind of bugs hide, and it is: 
* Duplicated in this method. 
* Hidden in this class: if I want to "simulate" a sensor measurement from another function/class, I need to reimplement it. 
Why we don't put this code as a method in the `Accelerometer` class?